### PR TITLE
17 create auth service

### PR DESCRIPTION
**Pull request by Andylovesmangos**
Created AuthService that returns userinfo (an object implementing UserInfo interface) and username (string). More fields may be added onto this interface later on, but requires update of instantiation of user info object.

### CODE REVIEW

File: get5-web/ClientApp/src/app/home/home.component.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
 import { Component } from '@angular/core';
+import { AuthService } from '../services/auth.services';
```
Comment (BrandonRS) [inline get5-web/ClientApp/src/app/home/home.component.ts:2]: Remove these changes from this file. These changes were only for testing implementation.

File: get5-web/ClientApp/src/app/services/auth.services.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@Injectable({
+    providedIn: 'root',
+   })
+export class AuthService{
+
+    private auth_url : string = "/api/auth/";
+
+    constructor(private httpClient : HttpClient){}
+
+    getUser(){
```
Comment (BrandonRS) [inline get5-web/ClientApp/src/app/services/auth.services.ts:14]: Add return type

File: get5-web/ClientApp/src/app/services/auth.services.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        
+        this.httpClient.get(this.auth_url + "username", {responseType: 'text'}).subscribe(response => {
+
+          //response is already string
+          return response;
+        });
+    }
+
+    getUserInfo(){
```
Comment (BrandonRS) [inline get5-web/ClientApp/src/app/services/auth.services.ts:23]: Add return type

File: get5-web/ClientApp/src/app/services/auth.services.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+        this.httpClient.get(this.auth_url + "userInfo", {responseType: 'json'}).subscribe(response=> {
+
+
+          let info: UserInfo = { username: Object.values(Object.values(response)[5]).toString().split(",").join("") } as UserInfo;
+
+          console.log(info);
```
Comment (BrandonRS) [inline get5-web/ClientApp/src/app/services/auth.services.ts:30]: Remove this log and return info instead.

File: get5-web/ClientApp/src/app/services/auth.services.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+          return response;
+        });
+    }
+
+    getUserInfo(){
+
+        this.httpClient.get(this.auth_url + "userInfo", {responseType: 'json'}).subscribe(response=> {
+
+
+          let info: UserInfo = { username: Object.values(Object.values(response)[5]).toString().split(",").join("") } as UserInfo;
```
Comment (BrandonRS) [inline get5-web/ClientApp/src/app/services/auth.services.ts:28]: The point of the user_info interface was to allow [this](https://stackoverflow.com/questions/47499324/angular-5-models-httpclient-type-casting).
Not the best example, but you should be able to see what's going on.

File: get5-web/ClientApp/src/app/home/home.component.ts [hunk 6]
```diff
@@ -1,8 +1,15 @@
...
+
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
+  providers: []
 })
+
 export class HomeComponent {
+
+  constructor() {
```
Comment (BrandonRS) [inline get5-web/ClientApp/src/app/home/home.component.ts:12]: You call this "reverting the file"?

File: get5-web/ClientApp/src/app/services/user_info.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
+export interface UserInfo {
```
Comment (BrandonRS) [inline get5-web/ClientApp/src/app/services/user_info.ts:2]: Put this file in a folder named "models" at the same level as the services folder.

**Review by BrandonRS — APPROVED** [review]

**Review by YLiao98 — COMMENTED** [review]
In file home.component.ts, the providers field is unused.

**Review by BrandonRS — APPROVED** [review]

**Pull request merged**